### PR TITLE
Scavenger hunt generator

### PR DESCRIPTION
**Pull request by huw30**
Scavenger Hunt generation view and tag selection view.

generator

### CODE REVIEW

**Review by wumingx89 — APPROVED** [review]
🚢

File: Sweeper/ViewControllers/SHGenerateViewController.swift [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+        // TODO: make this transition look nicer
+        tagSelectorView.modalPresentationStyle = UIModalPresentationStyle.formSheet
+        tagSelectorView.modalTransitionStyle = UIModalTransitionStyle.partialCurl
+
+        self.addChildViewController(tagSelectorView)
+        UIView.animate(withDuration: 0.3) {
+            self.view.addSubview(tagSelectorView.view)
+        }
+        tagSelectorView.didMove(toParentViewController: self)
```
Comment (sokolikp) [inline Sweeper/ViewControllers/SHGenerateViewController.swift:48]: Do we need a willMove?

File: Sweeper/ViewControllers/SHGenerateViewController.swift [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+// MARK: tag selector view controller delegate
+extension SHGenerateViewController: TagSelectorViewControllerDelegate {
+    func tagSelected(tagSelectorViewController: TagSelectorViewController, didSelectTag tag: Tag?) {
+        if (tag != nil) && (self.selectedTags.count < maxTagOrStops) {
+            self.selectedTags.append(tag!)
+            tagSelectionCollectionView.reloadData()
+        }
+    }
+}
```
Comment (sokolikp) [inline Sweeper/ViewControllers/SHGenerateViewController.swift:146]: Do you want to add the collectionView delegates here for conformity?

File: Sweeper/Views/InfiniteScrollActivityView.swift [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    func stopAnimating() {
+        self.activityIndicatorView.stopAnimating()
+        self.isHidden = true
+    }
+
+    func startAnimating() {
+        self.isHidden = false
+        self.activityIndicatorView.startAnimating()
+    }
+}
```
Comment (sokolikp) [inline Sweeper/Views/InfiniteScrollActivityView.swift:45]: Can we reuse this with other controllers? That is - should we make the name more generic?

**Review by sokolikp — APPROVED** [review]
Awesome!

File: Sweeper/Views/InfiniteScrollActivityView.swift (hunk 3 above)
Comment (huw30) [reply Sweeper/Views/InfiniteScrollActivityView.swift:45]: It is reusable by other controllers. I copied this from Codepath's guide code, and I think it's already generic :)

**Comment by sokolikp** [comment]
Oh and also.. merge conflicts.

File: Sweeper/ViewControllers/SHGenerateViewController.swift (hunk 1 above)
Comment (huw30) [reply Sweeper/ViewControllers/SHGenerateViewController.swift:48]: yeah, probably need one :)

File: Sweeper/ViewControllers/SHGenerateViewController.swift (hunk 2 above)
Comment (huw30) [reply Sweeper/ViewControllers/SHGenerateViewController.swift:146]: what do you mean?

**Pull request merged**